driver/registry_default.go

```diff
@@ -317,10 +318,14 @@ func (m *RegistryDefault) CourierConfig() config.CourierConfigs {
 	return m.Config()
 }
 
-func (m *RegistryDefault) selfServiceStrategies() []interface{} {
+type ider interface {
+	ID() identity.CredentialsType
+}
+
+func (m *RegistryDefault) selfServiceStrategies() []any {
 	if len(m.selfserviceStrategies) == 0 {
-		m.selfserviceStrategies = []interface{}{
-			password2.NewStrategy(m),
+		m.selfserviceStrategies = []any{
```

Comment:
Why so complicated? Would it not be enough to have a `SetSelfServiceStrategies(overrides)` function?